src/runtime/runtime.cs

```diff
@@ -306,17 +315,16 @@ private static IntPtr Get_PyObject_NextNotImplemented()
             return iternext;
         }
 
-        internal static void Shutdown()
+        internal static void Shutdown(ShutdownMode mode)
```

Comment:
If you tend to support asymmetric mode different from the initialization, you should add more validations. e.g. Normal mode can't upgrade to soft mode, it didn't save the recovery information in the first place.

Reply:
Good point, I'll do that